api/v1alpha08/kogitoserverlessbuild_types.go

```diff
@@ -15,38 +15,96 @@
+	"k8s.io/apimachinery/pkg/runtime"
+)
 
-	"github.com/kiegroup/kogito-serverless-operator/container-builder/api"
+type BuildPhase string
+
+const (
+	// BuildPhaseNone --
+	BuildPhaseNone BuildPhase = ""
+	// BuildPhaseInitialization --
+	BuildPhaseInitialization BuildPhase = "Initialization"
+	// BuildPhaseScheduling --
+	BuildPhaseScheduling BuildPhase = "Scheduling"
+	// BuildPhasePending --
+	BuildPhasePending BuildPhase = "Pending"
+	// BuildPhaseRunning --
+	BuildPhaseRunning BuildPhase = "Running"
+	// BuildPhaseSucceeded --
+	BuildPhaseSucceeded BuildPhase = "Succeeded"
+	// BuildPhaseFailed --
+	BuildPhaseFailed BuildPhase = "Failed"
+	// BuildPhaseInterrupted --
+	BuildPhaseInterrupted BuildPhase = "Interrupted"
+	// BuildPhaseError --
+	BuildPhaseError BuildPhase = "Error"
 )
 
-// KogitoServerlessBuildSpec defines the desired state of KogitoServerlessBuild
+type BuildTemplate struct {
+	// Timeout defines the Build maximum execution duration.
+	// The Build deadline is set to the Build start time plus the Timeout duration.
+	// If the Build deadline is exceeded, the Build context is canceled,
+	// and its phase set to BuildPhaseFailed.
+	// +kubebuilder:validation:Format=duration
+	Timeout metav1.Duration `json:"timeout,omitempty"`
+	// Resources optional compute resource requirements for the builder
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+	// Arguments lists the command line arguments to send to the builder
+	Arguments []string `json:"arguments,omitempty"`
+}
+
+// KogitoServerlessBuildSpec an abstraction over the actual build process performed by the platform.
 type KogitoServerlessBuildSpec struct {
-	// Workflow's unique identifier
-	WorkflowId string `json:"workflowId,omitempty"`
-	// Image name
-	ImageName string `json:"imageName,omitempty"`
-	// Middlename of the pod
-	PodMiddleName string `json:"podMiddleName,omitempty"`
-	// ContainerFile content used for the build
-	ContainerFile []byte `json:"containerFile,omitempty"`
+	BuildTemplate `json:",inline"`
 }
 
 // KogitoServerlessBuildStatus defines the observed state of KogitoServerlessBuild
 type KogitoServerlessBuildStatus struct {
-	// Workflow's unique identifier
-	WorkflowId string `json:"workflowId,omitempty"`
-	// Current kaniko buildphase
-	BuildPhase api.BuildPhase `json:"buildPhase,omitempty"`
-	// Kaniko's build, used to ping the build to update the buildphase
-	Builder api.Build `json:"builder,omitempty"`
+	// The image tag for this build instance
+	ImageTag string `json:"imageTag,omitempty"`
+	// Current phase of the build
+	BuildPhase BuildPhase `json:"buildPhase,omitempty"`
+	// Last error found during build
+	Error string `json:"error,omitempty"`
+	// InnerBuild is a reference to an internal build object, which can be anything known only to internal builders.
+	// +kubebuilder:pruning:PreserveUnknownFields
+	InnerBuild runtime.RawExtension `json:"innerBuild,omitempty" patchStrategy:"replace"`
```

Comment:
It's not meant to be an unstructured one since we do not necessarily follow the Kubernetes Spec. The inner build is a TypedLocalReference or a ContainerBuilder object.